superset-frontend/cccs-viz/plugins/plugin-chart-hello-world/src/plugin/controlPanel.ts

```diff
@@ -122,10 +113,50 @@ function validateIP(v: unknown) {
     }
   }
 
-  return ('is expected to be an ip address or cidr');
+  return (' is expected to be an ip address or cidr');
 }
 
-
+// filters is an array of adhoc filter with the following attributes
+// f.subject is the column name for example SRC_PORT
+// f.comparator is the value being tested, it can be a single value for operators like !=, >, <= etc
+// or it can be an array of values for example when the IN or NOT IN operator is used.
+// state is the current state of the adhoc filter control it includes
+// a copy of the columns as defined in the dataset model
+function adhocFilterValidator(filters: unknown, state: ControlState) {
+  if (Array.isArray(filters)) {
+    for (let i = 0; i < filters.length; i++) {
+      const filter = filters[i];
+      // Find the corresponding column in the model
+      const column = state.columns.find((c: any) => c.column_name == filter.subject);
+      if (typeof column !== 'undefined' && typeof column.type !== 'undefined') {
+        // Currently supporting 2 types of columns
+        // IPV4
+        // IPV4 FILTER
+        if (column.type.includes('IPV4')) {
+          const v = filter.comparator;
+          // check single value
+          if (typeof v === 'string' && v.trim().length > 0) {
```

Comment:
Any error handling needed if it's none of the types in the if-else?

Reply:
if it's not a type we validate it just reports no errors, we will of course add more of these at a later date

Reply:
Ahhhhh gotcha. Thought this might've been IP specific.